include/fs/fs_interface.h

```diff
@@ -13,7 +13,14 @@ extern "C" {
 
 #if defined(CONFIG_FILE_SYSTEM_LITTLEFS)
 #define MAX_FILE_NAME 256
-#else /* FAT_FS */
+#elif defined(CONFIG_FAT_FILESYSTEM_ELM)
+#if defined(CONFIG_FS_FATFS_LFN)
+#define MAX_FILE_NAME CONFIG_FS_FATFS_MAX_LFN
+#endif /* CONFIG_FS_FATFS_LFN */
+#endif /* filesystem selection */
+
+#ifndef MAX_FILE_NAME
+/* Fallback to SFN when no filesystem is explicitly selected */
```

Comment:
This is not right.  If you have FatFS, but didn't pick LFN, you should get 8.3 SFN.  Not whatever default is provided when the file system isn't defined.  Please provide the fallback for `FAT_FILESYSTEM_ELM` within the branch for that filesystem.

If you want to add an `#else /* filesystem selection*/` that also defines `MAX_FILE_NAME` that'd be fine.  It's not necessary because the subsystem won't work in that case until #25883 is fixed, though.

Also the fix(es) should be squashed into the original commit that changed the definition.

Reply:
Before I committed, I ran `$ZEPHYR_BASE/scripts/sanitycheck -T tests/subsys/fs/` with the following:
```c
#if defined(CONFIG_FILE_SYSTEM_LITTLEFS)
#define MAX_FILE_NAME 256
#elif defined(CONFIG_FAT_FILESYSTEM_ELM)
#if defined(CONFIG_FS_FATFS_LFN)
#define MAX_FILE_NAME CONFIG_FS_FATFS_MAX_LFN
#else
#define MAX_FILE_NAME 12 /* Uses 8.3 SFN */
#endif /* CONFIG_FS_FATFS_LFN */
#endif /* filesystem selection */
```

sanity check failed with the following:
```
INFO    - JOBS: 4
INFO    - Selecting default platforms per test case
INFO    - Building initial testcase list...
INFO    - 11 test configurations selected, 1981 configurations discarded due to filters.
INFO    - Adding tasks to the queue...

ERROR   - native_posix              tests/subsys/fs/fcb/filesystem.fcb                 FAILED: Build failure
ERROR   - see: /home/nzsmartie/Projects/zephyr/zephyr/sanity-out/native_posix/tests/subsys/fs/fcb/filesystem.fcb/build.log
INFO    - Total complete:    1/  11   9%  skipped:    0, failed:    1
ERROR   - qemu_x86                  tests/subsys/fs/nvs/filesystem.nvs                 FAILED: Build failure
ERROR   - see: /home/nzsmartie/Projects/zephyr/zephyr/sanity-out/qemu_x86/tests/subsys/fs/nvs/filesystem.nvs/build.log
INFO    - Total complete:   11/  11  100%  skipped:    0, failed:    2
INFO    - 9 of 11 tests passed (81.82%), 2 failed, 0 skipped with 0 warnings in 74.84 seconds
INFO    - In total 8 test cases were executed on 2 out of total 250 platforms (0.80%)
INFO    - 9 tests executed on platforms, 2 tests were only built.
```
The first build log contained:
```
[ 27%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
In file included from /home/nzsmartie/Projects/zephyr/zephyr/include/sys/fdtable.h:12,
                 from /home/nzsmartie/Projects/zephyr/zephyr/lib/os/fdtable.c:19:
/home/nzsmartie/Projects/zephyr/zephyr/include/fs/fs.h:91:12: error: ‘MAX_FILE_NAME’ undeclared here (not in a function)
   91 |  char name[MAX_FILE_NAME + 1];
```
while the second also contained:
```
[ 16%] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
In file included from /home/nzsmartie/Projects/zephyr/zephyr/include/sys/fdtable.h:12,
                 from /home/nzsmartie/Projects/zephyr/zephyr/lib/os/fdtable.c:19:
/home/nzsmartie/Projects/zephyr/zephyr/include/fs/fs.h:91:12: error: 'MAX_FILE_NAME' undeclared here (not in a function); did you mean 'MK_ISR_NAME'?
   91 |  char name[MAX_FILE_NAME + 1];
      |            ^~~~~~~~~~~~~
      |            MK_ISR_NAME
```

My intuition is to preserve the original behaviour of having `MAX_FILE_NAME` defined as `12` (8.3) since these testcases apper to use neither FatFS or LittleFS but still rely on `MAX_FILE_NAME`

By having the fallback `#ifndef`, i'm not repeating code and preserve readability.

Reply:
I do agree with the squashing, and will do that once changes are approved

Reply:
> By having the fallback `#ifndef`, i'm not repeating code and preserve readability.

No, you're not.  You're leaving the specification for `FILESYSTEM_FAT_ELM` incomplete, then using its unhandled case as a default when there are no file systems.  We may want to change the default for no-file-system to something else without breaking FatFS.

Please make the change as I requested.  It should look like this:
```
#if defined(CONFIG_FILE_SYSTEM_LITTLEFS)
#define MAX_FILE_NAME 256
#elif defined(CONFIG_FAT_FILESYSTEM_ELM)
#if defined(CONFIG_FS_FATFS_LFN)
#define MAX_FILE_NAME CONFIG_FS_FATFS_MAX_LFN
#else /* CONFIG_FS_FATFS_LFN */
#define MAX_FILE_NAME 12 /* Uses 8.3 SFN */
#endif /* CONFIG_FS_FATFS_LFN */
#else /* filesystem selection */
/* Use standard 8.3 when no filesystem is explicitly selected */
#define MAX_FILE_NAME 12
#endif /* filesystem selection */
```

Reply:
You should squash first; it's not going to get approved as separate commits that need to be squashed.

Reply:
That makes sense. Sorry for the delay on the update. Changes are pushed and thank you for your feedback!